### ISSUE TITLE
Use ASP NET Core's `AuthenticationScheme` to handle `AbpAuthorizationException`.

### ISSUE THREAD

**Issue by maliming** [abpframework/abp#9926]
Related #2643 #3848  #4785  #5235 #7503

> When you call the controller that requires authentication, the authentication middleware finds that the current user is not authenticated and calls ChallengeAsync (DefaultChallengeScheme is identity cookie). At this point the request has been shorted.
> If your anonymous controller calls the application service method, it will execute the ABP filter and interceptor. The framework throws AbpAuthorizationException, the filter wraps the exception into a 401, etc.

We wrap the exception in `ExceptionFilter` as `RemoteServiceErrorResponse` and specify the HTTP status code for it.

An exception will be throw after the permission check fails. It will not get an exception in ASP NET Core but is handled by the authentication scheme.

I think we should let the authentication scheme handle `AbpAuthorizationException`, This will make the above behavior consistent.

---

Update:

Cookies authentication will set 401 for AJAX requests.

https://github.com/dotnet/aspnetcore/blob/v6.0.0/src/Security/Authentication/Cookies/src/CookieAuthenticationEvents.cs#L46-L50

### Forward any requests that start with /api to the JWT scheme.

```cs
context.Services.ConfigureApplicationCookie(options =>
{
    // Forward any requests that start with /api to the JWT scheme.
    options.ForwardDefaultSelector = ctx =>
    {
        return ctx.Request.Path.StartsWithSegments("/api") ? JwtBearerDefaults.AuthenticationScheme : null;
    };
});

context.Services.AddAuthentication()
    .AddJwtBearer(options =>
    {
        options.Authority = configuration["AuthServer:Authority"];
        //...
    });
```

**Comment by maliming** [comment]
```cs
protected virtual async Task HandleAndWrapException(ExceptionContext context)
{
    var exceptionHandlingOptions = context.GetRequiredService<IOptions<AbpExceptionHandlingOptions>>().Value;
    var exceptionToErrorInfoConverter = context.GetRequiredService<IExceptionToErrorInfoConverter>();
    var remoteServiceErrorInfo  = exceptionToErrorInfoConverter.Convert(context.Exception, exceptionHandlingOptions.SendExceptionsDetailsToClients);

    var logLevel = context.Exception.GetLogLevel();

    var remoteServiceErrorInfoBuilder = new StringBuilder();
    remoteServiceErrorInfoBuilder.AppendLine($"---------- {nameof(RemoteServiceErrorInfo)} ----------");
    remoteServiceErrorInfoBuilder.AppendLine(context.GetRequiredService<IJsonSerializer>().Serialize(remoteServiceErrorInfo, indented: true));

    var logger = context.GetService<ILogger<AbpExceptionFilter>>(NullLogger<AbpExceptionFilter>.Instance);

    logger.LogWithLevel(logLevel, remoteServiceErrorInfoBuilder.ToString());

    logger.LogException(context.Exception, logLevel);

    await context.GetRequiredService<IExceptionNotifier>().NotifyAsync(new ExceptionNotificationContext(context.Exception));

    if (context.Exception is AbpAuthorizationException)
    {
        var isAuthenticated = context.HttpContext.RequestServices.GetRequiredService<ICurrentUser>().IsAuthenticated;
        var abpExceptionOptions = context.HttpContext.RequestServices.GetRequiredService<IOptions<AbpExceptionOptions>>().Value;
        if (abpExceptionOptions.UseAuthenticationScheme)
        {
            var handlers = context.HttpContext.RequestServices.GetRequiredService<IAuthenticationHandlerProvider>();
            var handler = await handlers.GetHandlerAsync(context.HttpContext, abpExceptionOptions.AuthenticationScheme);
            if (handler != null)
            {
                if (isAuthenticated)
                {
                    await handler.ForbidAsync(new AuthenticationProperties());
                }
                else
                {
                    await handler.ChallengeAsync(new AuthenticationProperties());
                }
                context.Exception = null; //Handled!
                return;
            }

            var authenticationSchemeProvider = context.HttpContext.RequestServices.GetRequiredService<IAuthenticationSchemeProvider>();
            if (context.HttpContext.RequestServices.GetRequiredService<ICurrentUser>().IsAuthenticated)
            {
                var scheme = isAuthenticated
                    ? await authenticationSchemeProvider.GetDefaultForbidSchemeAsync()
                    : await authenticationSchemeProvider.GetDefaultChallengeSchemeAsync();

                if (scheme != null)
                {
                    handler = await handlers.GetHandlerAsync(context.HttpContext, scheme.Name);
                    if (handler != null)
                    {
                        if (isAuthenticated)
                        {
                            await handler.ForbidAsync(new AuthenticationProperties());
                        }
                        else
                        {
                            await handler.ChallengeAsync(new AuthenticationProperties());
                        }
                        context.Exception = null; //Handled!
                        return;
                    }
                }
            }
        }
    }

    context.HttpContext.Response.Headers.Add(AbpHttpConsts.AbpErrorFormat, "true");
    context.HttpContext.Response.StatusCode = (int) context
        .GetRequiredService<IHttpExceptionStatusCodeFinder>()
        .GetStatusCode(context.HttpContext, context.Exception);

    context.Result = new ObjectResult(new RemoteServiceErrorResponse(remoteServiceErrorInfo));

    context.Exception = null; //Handled!
}
```

**Comment by maliming** [comment]
Cookies authentication will set 401 for AJAX requests.

https://github.com/dotnet/aspnetcore/blob/v6.0.0/src/Security/Authentication/Cookies/src/CookieAuthenticationEvents.cs#L46-L50

**Comment by sahvey** [comment]
```
context.Services.ConfigureApplicationCookie(options =>
{
    // Forward any requests that start with /api to the JWT scheme.
    options.ForwardDefaultSelector = ctx =>
    {
        return ctx.Request.Path.StartsWithSegments("/api") ? JwtBearerDefaults.AuthenticationScheme : null;
    };
});

context.Services.AddAuthentication()
    .AddJwtBearer(options =>
    {
        options.Authority = configuration["AuthServer:Authority"];
        //...
    });
```

When i set it as above, when i use postman or swaggere, only Error codes such as 401 or 403 are returned as seen below. 
![Error2](https://user-images.githubusercontent.com/46441777/170256697-154bb1f6-8603-4100-8c5c-3c8ed2c76382.png)

However, as seen below, an error should be returned according to the 401 or 403 error message template. 
![Error1](https://user-images.githubusercontent.com/46441777/170256615-11539ab6-1b14-4e09-9dcd-e05d839c930b.png)

In other words, the solution you specified does not completely solve our problem.
